src/services/API/RootClient.js

```diff
@@ -77,34 +81,59 @@ client.interceptors.response.use(
 
     // Handle expired/deleted access token here
     if (status === 401) {
-      // reauthenticate the user, and until that is happenining, pause all other
-      // requests. Once the user is authenticated, release all the paused requests
-      // with the new token attached to the header
-      return UserFunctionalService.reAuthenticate(store)
-        .then(() => {
-          if (store.state.auth.accessToken != null) {
-            var newAccessToken = store.state.auth.accessToken.access_token;
-            // Add the new access token to the header of the request
-            error.config.headers["Authorization"] = `Bearer ${newAccessToken}`;
-
+      // reAuthenticate the user if this is the first reAuthentication call. While reAuthentication is going on
+      // if some other request ends up here, then wait for the reAuthentication process to finish. Once that is done,
+      // set the newly retrieved access token as an auth header and retry the request. Any errors in the reauthentication
+      // process are caught and the user is logged out.
+      let run = async () => {
+        try {
+          let reAuthenticationState = store.state.auth.reAuthenticationState;
+          switch (reAuthenticationState) {
+            case "dropped":
+              // re authentication process was dropped, throw an error.
```

Comment:
in some places, reAuthentication is written, in some places re authentication is written. Let's stick to one. I think re-authentication is fine.

Reply:
```suggestion
              // re-authentication process was dropped, throw an error.
```